hw-model/tests/model_tests.rs

```diff
@@ -271,3 +271,117 @@ fn test_pcr_extend() {
 
     model.step_until_exit_success().unwrap();
 }
+
+#[test]
+fn test_mbox_uc_response_dlen_after_data() {
+    let mut model = run_fw_elf_with_rand_puf(
+        &caliptra_builder::build_firmware_elf(&firmware::hw_model_tests::MAILBOX_RESPONDER)
+            .unwrap(),
+    );
+    let mbox = model.soc_mbox();
+    // Lock mailbox
+    assert!(!mbox.lock().read().lock());
+    assert!(mbox.lock().read().lock());
+
+    // This command writes 3 words to dint but doesn't update dlen immediately
+    mbox.cmd().write(|_| 0x1000_3000);
+    mbox.dlen().write(|_| 4);
+    mbox.datain().write(|_| 0x3a48_735b);
+    mbox.execute().write(|w| w.execute(true));
+
+    while model.soc_mbox().status().read().status().cmd_busy() {
+        model.step();
+    }
+    let mbox = model.soc_mbox();
+    assert_eq!(12, mbox.dlen().read());
+    assert_eq!(0xdf06_065d, mbox.dataout().read());
+    assert_eq!(0x2926_a95d, mbox.dataout().read());
+    assert_eq!(0x55ca_5482, mbox.dataout().read());
+}
+
+#[test]
+fn test_mbox_uc_response_dlen_never_written() {
+    let mut model = run_fw_elf_with_rand_puf(
+        &caliptra_builder::build_firmware_elf(&firmware::hw_model_tests::MAILBOX_RESPONDER)
+            .unwrap(),
+    );
+    let mbox = model.soc_mbox();
+    // Lock mailbox
+    assert!(!mbox.lock().read().lock());
+    assert!(mbox.lock().read().lock());
+
+    // This command writes two words to din but doesn't set dlen
+    mbox.cmd().write(|_| 0x1000_3001);
+    mbox.dlen().write(|_| 12);
+    mbox.datain().write(|_| 0x508b_7a98);
+    mbox.datain().write(|_| 0x9dbe_d0ec);
+    mbox.datain().write(|_| 0x3ef7_9f17);
```

Comment:
Same, let's add a comment for why the caller is writing to the mailbox

Reply:
Done.